### ISSUE TITLE
Error in the sampling process if regression does not include certain parameter 

### ISSUE THREAD

**Issue by Jasonleng** [lnccbrown/HSSM#219]
When I am running the example code in the tutorial:
```
model_reg_v_angle_hier = hssm.HSSM(
    data=sub_data,
    model="angle",
    include=[
        {
        "name": "v",
        "prior": {
                  "Intercept": {"name": "Uniform", 
                                "lower": -3.0, "upper": 3.0, 
                                "initval": 0.0},
                 },
        "formula": "v ~ 1 + (1|subj_idx)",
        "link": "identity",
        }
    ],
)
```
It works fine and I got the model fitted. However, when I change the parameter I want to fit the regression
```
model_reg_v_angle_hier = hssm.HSSM(
    data=sub_data,
    model="angle",
    include=[
        {
        "name": "a",
        "prior": {
                  "Intercept": {"name": "Uniform", 
                                "lower": -3.0, "upper": 3.0, 
                                "initval": 0.0},
                 },
        "formula": "a ~ 1 + (1|subj_idx)",
        "link": "identity",
        }
    ],
)
```
with this sample code:from jax.config import config
```
config.update("jax_enable_x64", False)
model_reg_v_angle_hier.sample(sampler="nuts_numpyro", 
                              chains = 2, 
                              cores = 1, 
                              draws = 1000,
                              tune = 1000)
```
I will get this error:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/jax/_src/api.py:1260, in _mapped_axis_size.<locals>._get_axis_size(name, shape, axis)
   1259 try:
-> 1260   return shape[axis]
   1261 except (IndexError, TypeError) as e:

IndexError: tuple index out of range

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pytensor/compile/function/types.py:970, in Function.__call__(self, *args, **kwargs)
    968 try:
    969     outputs = (
--> 970         self.vm()
    971         if output_subset is None
    972         else self.vm(output_subset=output_subset)
    973     )
    974 except Exception:

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pytensor/graph/op.py:543, in Op.make_py_thunk.<locals>.rval(p, i, o, n, params)
    539 @is_thunk_type
    540 def rval(
    541     p=p, i=node_input_storage, o=node_output_storage, n=node, params=None
    542 ):
--> 543     r = p(n, [x[0] for x in i], o)
    544     for o in node.outputs:

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/hssm/distribution_utils/onnx/onnx.py:175, in make_jax_logp_ops.<locals>.LANLogpOp.perform(self, node, inputs, output_storage)
    161 """Perform the Apply node.
    162 
    163 Parameters
   (...)
    173     the Op.
    174 """
--> 175 result = logp(*inputs)
    176 output_storage[0][0] = np.asarray(result, dtype=node.outputs[0].dtype)

    [... skipping hidden 18 frame]

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/jax/_src/api.py:1264, in _mapped_axis_size.<locals>._get_axis_size(name, shape, axis)
   1263 # TODO(mattjj): better error message here
-> 1264 raise ValueError(
   1265     f"{name} was requested to map its argument along axis {axis}, "
   1266     f"which implies that its rank should be at least {min_rank}, "
   1267     f"but is only {len(shape)} (its shape is {shape})") from e

ValueError: vmap was requested to map its argument along axis 0, which implies that its rank should be at least 1, but is only 0 (its shape is ())

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
Cell In[41], line 3
      1 from jax.config import config
      2 config.update("jax_enable_x64", False)
----> 3 model_reg_v_angle_hier.sample(sampler="nuts_numpyro", 
      4                               chains = 2, 
      5                               cores = 1, 
      6                               draws = 1000,
      7                               tune = 1000)

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/hssm/hssm.py:463, in HSSM.sample(self, sampler, **kwargs)
    460     if "step" not in kwargs:
    461         kwargs["step"] = pm.Slice(model=self.pymc_model)
--> 463 self._inference_obj = self.model.fit(inference_method=sampler, **kwargs)
    465 return self.traces

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/bambi/models.py:325, in Model.fit(self, draws, tune, discard_tuned_samples, omit_offsets, include_mean, inference_method, init, n_init, chains, cores, random_seed, **kwargs)
    318     response = self.components[self.response_name]
    319     _log.info(
    320         "Modeling the probability that %s==%s",
    321         response.response_term.name,
    322         str(response.response_term.success),
    323     )
--> 325 return self.backend.run(
    326     draws=draws,
    327     tune=tune,
    328     discard_tuned_samples=discard_tuned_samples,
    329     omit_offsets=omit_offsets,
    330     include_mean=include_mean,
    331     inference_method=inference_method,
    332     init=init,
    333     n_init=n_init,
    334     chains=chains,
    335     cores=cores,
    336     random_seed=random_seed,
    337     **kwargs,
    338 )

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/bambi/backend/pymc.py:96, in PyMCModel.run(self, draws, tune, discard_tuned_samples, omit_offsets, include_mean, inference_method, init, n_init, chains, cores, random_seed, **kwargs)
     94 # NOTE: Methods return different types of objects (idata, approximation, and dictionary)
     95 if inference_method in ["mcmc", "nuts_numpyro", "nuts_blackjax"]:
---> 96     result = self._run_mcmc(
     97         draws,
     98         tune,
     99         discard_tuned_samples,
    100         omit_offsets,
    101         include_mean,
    102         init,
    103         n_init,
    104         chains,
    105         cores,
    106         random_seed,
    107         inference_method,
    108         **kwargs,
    109     )
    110 elif inference_method == "vi":
    111     result = self._run_vi(**kwargs)

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/bambi/backend/pymc.py:211, in PyMCModel._run_mcmc(self, draws, tune, discard_tuned_samples, omit_offsets, include_mean, init, n_init, chains, cores, random_seed, sampler_backend, **kwargs)
    208     if not chains:
    209         # sample_numpyro_nuts does not handle chains = None like pm.sample does
    210         chains = 4
--> 211     idata = pymc.sampling_jax.sample_numpyro_nuts(
    212         draws=draws,
    213         tune=tune,
    214         chains=chains,
    215         random_seed=random_seed,
    216         **kwargs,
    217     )
    218 elif sampler_backend == "nuts_blackjax":
    219     import pymc.sampling_jax  # pylint: disable=import-outside-toplevel

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/sampling/jax.py:625, in sample_numpyro_nuts(draws, tune, chains, target_accept, random_seed, initvals, model, var_names, progressbar, keep_untransformed, chain_method, postprocessing_backend, postprocessing_chunks, idata_kwargs, nuts_kwargs, **kwargs)
    622 tic1 = datetime.now()
    623 print("Compiling...", file=sys.stdout)
--> 625 init_params = _get_batched_jittered_initial_points(
    626     model=model,
    627     chains=chains,
    628     initvals=initvals,
    629     random_seed=random_seed,
    630 )
    632 logp_fn = get_jaxified_logp(model, negative_logp=False)
    634 nuts_kwargs = _update_numpyro_nuts_kwargs(nuts_kwargs)

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/sampling/jax.py:247, in _get_batched_jittered_initial_points(model, chains, initvals, random_seed, jitter, jitter_max_retries)
    230 def _get_batched_jittered_initial_points(
    231     model: Model,
    232     chains: int,
   (...)
    236     jitter_max_retries: int = 10,
    237 ) -> Union[np.ndarray, List[np.ndarray]]:
    238     """Get jittered initial point in format expected by NumPyro MCMC kernel
    239 
    240     Returns
   (...)
    244         Each item has shape `(chains, *var.shape)`
    245     """
--> 247     initial_points = _init_jitter(
    248         model,
    249         initvals,
    250         seeds=_get_seeds_per_chain(random_seed, chains),
    251         jitter=jitter,
    252         jitter_max_retries=jitter_max_retries,
    253     )
    254     initial_points_values = [list(initial_point.values()) for initial_point in initial_points]
    255     if chains == 1:

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/sampling/mcmc.py:1221, in _init_jitter(model, initvals, seeds, jitter, jitter_max_retries)
   1219 if i < jitter_max_retries:
   1220     try:
-> 1221         model.check_start_vals(point)
   1222     except SamplingError:
   1223         # Retry with a new seed
   1224         seed = rng.randint(2**30, dtype=np.int64)

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/model.py:1781, in Model.check_start_vals(self, start)
   1775     valid_keys = ", ".join(value_names_set)
   1776     raise KeyError(
   1777         "Some start parameters do not appear in the model!\n"
   1778         f"Valid keys are: {valid_keys}, but {extra_keys} was supplied"
   1779     )
-> 1781 initial_eval = self.point_logps(point=elem)
   1783 if not all(np.isfinite(v) for v in initial_eval.values()):
   1784     raise SamplingError(
   1785         "Initial evaluation of model at starting point failed!\n"
   1786         f"Starting values:\n{elem}\n\n"
   1787         f"Logp initial evaluation results:\n{initial_eval}\n"
   1788         "You can call `model.debug()` for more details."
   1789     )

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/model.py:1816, in Model.point_logps(self, point, round_vals)
   1810 factors = self.basic_RVs + self.potentials
   1811 factor_logps_fn = [pt.sum(factor) for factor in self.logp(factors, sum=False)]
   1812 return {
   1813     factor.name: np.round(np.asarray(factor_logp), round_vals)
   1814     for factor, factor_logp in zip(
   1815         factors,
-> 1816         self.compile_fn(factor_logps_fn)(point),
   1817     )
   1818 }

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/pytensorf.py:763, in PointFunc.__call__(self, state)
    762 def __call__(self, state):
--> 763     return self.f(**state)

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pytensor/compile/function/types.py:983, in Function.__call__(self, *args, **kwargs)
    981     if hasattr(self.vm, "thunks"):
    982         thunk = self.vm.thunks[self.vm.position_of_error]
--> 983     raise_with_op(
    984         self.maker.fgraph,
    985         node=self.vm.nodes[self.vm.position_of_error],
    986         thunk=thunk,
    987         storage_map=getattr(self.vm, "storage_map", None),
    988     )
    989 else:
    990     # old-style linkers raise their own exceptions
    991     raise

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pytensor/link/utils.py:535, in raise_with_op(fgraph, node, thunk, exc_info, storage_map)
    530     warnings.warn(
    531         f"{exc_type} error does not allow us to add an extra error message"
    532     )
    533     # Some exception need extra parameter in inputs. So forget the
    534     # extra long error message in that case.
--> 535 raise exc_value.with_traceback(exc_trace)

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pytensor/compile/function/types.py:970, in Function.__call__(self, *args, **kwargs)
    967 t0_fn = time.perf_counter()
    968 try:
    969     outputs = (
--> 970         self.vm()
    971         if output_subset is None
    972         else self.vm(output_subset=output_subset)
    973     )
    974 except Exception:
    975     restore_defaults()

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/pytensor/graph/op.py:543, in Op.make_py_thunk.<locals>.rval(p, i, o, n, params)
    539 @is_thunk_type
    540 def rval(
    541     p=p, i=node_input_storage, o=node_output_storage, n=node, params=None
    542 ):
--> 543     r = p(n, [x[0] for x in i], o)
    544     for o in node.outputs:
    545         compute_map[o][0] = True

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/hssm/distribution_utils/onnx/onnx.py:175, in make_jax_logp_ops.<locals>.LANLogpOp.perform(self, node, inputs, output_storage)
    160 def perform(self, node, inputs, output_storage):
    161     """Perform the Apply node.
    162 
    163     Parameters
   (...)
    173         the Op.
    174     """
--> 175     result = logp(*inputs)
    176     output_storage[0][0] = np.asarray(result, dtype=node.outputs[0].dtype)

    [... skipping hidden 18 frame]

File ~/anaconda/HSSMEnv/lib/python3.9/site-packages/jax/_src/api.py:1264, in _mapped_axis_size.<locals>._get_axis_size(name, shape, axis)
   1262 min_rank = axis + 1 if axis >= 0 else -axis
   1263 # TODO(mattjj): better error message here
-> 1264 raise ValueError(
   1265     f"{name} was requested to map its argument along axis {axis}, "
   1266     f"which implies that its rank should be at least {min_rank}, "
   1267     f"but is only {len(shape)} (its shape is {shape})") from e

ValueError: vmap was requested to map its argument along axis 0, which implies that its rank should be at least 1, but is only 0 (its shape is ())
Apply node that caused the error: LANLogpOp(rt,response{[[0.681 1. ... 78 1.   ]]}, Composite{...}.1, Composite{...}.0, Composite{...}.1, Composite{...}.1, Composite{...}.1)
Toposort index: 16
Inputs types: [TensorType(float32, shape=(28785, 2)), TensorType(float32, shape=()), TensorType(float32, shape=(28785,)), TensorType(float32, shape=()), TensorType(float32, shape=()), TensorType(float32, shape=())]
Inputs shapes: [(28785, 2), (), (28785,), (), (), ()]
Inputs strides: [(8, 4), (), (4,), (), (), ()]
Inputs values: ['not shown', array(1.3105308, dtype=float32), 'not shown', array(0.64052427, dtype=float32), array(0.9996157, dtype=float32), array(0.4004177, dtype=float32)]
Outputs clients: [[Switch(BroadcastTo.0, [-66.1], LANLogpOp.0), Shape_i{0}(LANLogpOp.0)]]

Backtrace when the node is created (use PyTensor flag traceback__limit=N to make it longer):
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/model.py", line 764, in logp
    rv_logps = transformed_conditional_logp(
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/logprob/basic.py", line 366, in transformed_conditional_logp
    temp_logp_terms = conditional_logp(
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/logprob/basic.py", line 295, in conditional_logp
    q_logprob_vars = _logprob(
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/functools.py", line 888, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/site-packages/pymc/distributions/distribution.py", line 130, in logp
    return class_logp(value, *dist_params)
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/site-packages/hssm/distribution_utils/dist.py", line 279, in logp
    logp = loglik(data, *dist_params)
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/site-packages/pytensor/graph/op.py", line 295, in __call__
    node = self.make_node(*inputs, **kwargs)
  File "/users/xleng/anaconda/HSSMEnv/lib/python3.9/site-packages/hssm/distribution_utils/onnx/onnx.py", line 156, in make_node
    outputs = [pt.vector()]

HINT: Use the PyTensor flag `exception_verbosity=high` for a debug print-out and storage map footprint of this Apply node.
```
I think this could be due to the fact that I did not include variables in the regression following the order of parameters in the list_params. When I include both v and a in the regression:
```
model_reg_v_angle_hier = hssm.HSSM(
    data=sub_data,
    model="angle",
    include=[
        {
        "name": "a",
        "prior": {
                  "Intercept": {"name": "Uniform", 
                                "lower": 1, "upper": 3.0, 
                                "initval": 1.0},
                 },
        "formula": "a ~ 1 + (1|subj_idx)",
        "link": "identity",
        },
        {
        "name": "v",
        "prior": {
                  "Intercept": {"name": "Uniform", 
                                "lower": -3, "upper": 3.0, 
                                "initval": 0},
                 },
        "formula": "v ~ 1 + (1|subj_idx)",
        "link": "identity",
        }
    ],
)
```
The sampling works fine.

**Comment by frankmj** [comment]
This is likely because you have a prior for a in there that ranges from -3 to 3 and negative values of a are not possible. (more generally it will likely give this error for any prior that includes parameter ranges outside of those trained by the LAN).

I suspect this is just because you copy and pasted the code from your v regression and forgot to change the prior - but this also relates to my suggestion in another issue that we remove priors from these regressions unless the user wants to override the default.

**Comment by Jasonleng** [comment]
> This is likely because you have a prior for a in there that ranges from -3 to 3 and negative values of a are not possible. (more generally it will likely give this error for any prior that includes parameter ranges outside of those trained by the LAN).
> 
> I suspect this is just because you copy and pasted the code from your v regression and forgot to change the prior - but this also relates to my suggestion in another issue that we remove priors from these regressions unless the user wants to override the default.

Yes I realized that and then I tried this:
```
model_reg_v_angle_hier = hssm.HSSM(
    data=sub_data,
    model="angle",
    include=[
        {
        "name": "a",
        "prior": {
                  "Intercept": {"name": "Uniform", 
                                "lower": 1.0, "upper": 3.0, 
                                "initval": 2.0},
                 },
        "formula": "a ~ 1 + (1|subj_idx)",
        "link": "identity",
        }
    ],
)

model_reg_v_angle_hier.sample(sampler="nuts_numpyro", 
                              chains = 2, 
                              cores = 1, 
                              draws = 1000,
                              tune = 1000)
```

And I still get the same error.

**Comment by digicosmos86** [comment]
This might be due to a bug in how parameters with different dimensions are handled in the case of `approx_differentiable` likelihoods. We will flag this as a bug and work on fixing this in the next release

**Comment by frankmj** [comment]
did you make sure that 3 is allowed in the bounds of the LAN? probably is but just checking

**Comment by digicosmos86** [comment]
For `angle` models the bounds are `(0.3, 3.0)`. In the non-regression case we just truncate the prior distribution according to the bounds. In the regression case, we make sure that the out-of-bound values are discarded

**Comment by digicosmos86** [comment]
Hi @Jasonleng! This should be fixed now in HSSM v0.1.2, which is coming out in a few hours. Please feel free to update hssm and try it out :)

**Comment by Jasonleng** [comment]
> Hi @Jasonleng! This should be fixed now in HSSM v0.1.2, which is coming out in a few hours. Please feel free to update hssm and try it out :)

Great. Thanks Paul! I will definitely try it out.
